Sources/Db/APIs/MySQL.php

```diff
@@ -2584,11 +2584,17 @@ protected function replacement__callback(array $matches): string
 			return '\'' . mysqli_real_escape_string($this->temp_connection, $matches[2]) . '\'';
 		}
 
-		if (!isset($this->temp_values[$matches[2]])) {
+		if (!array_key_exists($matches[2], $this->temp_values)) {
 			$this->error_backtrace('The database value you\'re trying to insert does not exist: ' . Utils::htmlspecialchars($matches[2]), '', E_USER_ERROR, __FILE__, __LINE__);
 		}
 
-		$replacement = $this->temp_values[$matches[2]];
+		$replacement = $this->temp_values[$matches[2]] ?? null;
+
+		if ($replacement === null && str_starts_with($matches[1], 'array_')) {
+			$this->error_backtrace('The database value you\'re trying to insert does not exist: ' . Utils::htmlspecialchars($matches[2]), '', E_USER_ERROR, __FILE__, __LINE__);
+		} elseif ($replacement === null) {
+			return 'NULL';
+		}
```

Comment:
```suggestion
		$replacement = $this->temp_values[$matches[2]];

		if ($replacement === null) {
			if (str_starts_with($matches[1], 'array_')) {
				$this->error_backtrace('The database value you\'re trying to insert does not exist: ' . Utils::htmlspecialchars($matches[2]), '', E_USER_ERROR, __FILE__, __LINE__);
			}
			
			return 'NULL';
		}
```

Don't need the null coalescing operator since `array_key_exists()` was used above.